internal/data/inmemory/carts/carts.go

```diff
@@ -0,0 +1,55 @@
+package carts
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	errors "github.com/Red-Sock/trace-errors"
+
+	"github.com/Red-Sock/Red-Cart/internal/domain/cart"
+)
+
+type Carts struct {
+	idCart atomic.Int64
+	rw     sync.RWMutex
+	m      map[int64]cart.Cart
+}
+
+// Точно не пустой!
+func New() *Carts {
+	return &Carts{
+		m: make(map[int64]cart.Cart),
+	}
+}
+
+func (c *Carts) Get(ctx context.Context, idOwner int64) (cart.Cart, error) {
+	c.rw.RLock()
+	defer c.rw.RUnlock()
+	cart, ok := c.m[idOwner]
+
+	if ok {
+		msg := fmt.Sprintf("У вас уже есть корзина с идентификатором = %d", cart.Id)
+		return c.m[idOwner], errors.New(msg)
```

Comment:
В будущем МРе подправишь
```suggestion
	cart, ok := c.m[idOwner]
	if ok {
		msg := fmt.Sprintf("У вас уже есть корзина с идентификатором = %d", cart.Id)
		return c.m[idOwner], errors.New(msg)
```